content_scripts/vimium_frontend.coffee

```diff
@@ -133,6 +133,28 @@ class GrabBackFocus extends Mode
     element.blur()
     @suppressEvent
 
+# Pages can load new content dynamically and change the displayed URL using history.pushState. Since this can
+# often be indistinguishable from an actual new page load for the user, we should also re-start GrabBackFocus
+# for these as well.
+handlerStack.push
+  click: (event) ->
+    target = event.target
+    while target
+      # Often, a link which triggers a content load and url change with javascript will also have the new
+      # url as it's href attribute.
+      if target.tagName == "A" and
```

Comment:
I like this approach, @mrmr1993.

Four comments:
1. Could you add a reference to the original issues?
2. Elsewhere, we use `toLowerCase` for comparisons like this.  Do we need that here too?
3. If you add a `_name` property, we get better debugging in the handler stack.
4. Is it possible to have an input embedded within an element which satisfies the condition here?  Something like:

```HTML
<a> .... <input> ... </input> </a>
```